storm-client/src/jvm/org/apache/storm/utils/ObjectReader.java

```diff
@@ -76,6 +76,32 @@ public static Integer getInt(Object o, Integer defaultValue) {
         throw new IllegalArgumentException("Don't know how to convert " + o + " to int");
     }
 
+    public static Long getLong(Object o) {
+        return getLong(o, null);
+    }
+
+    public static Long getLong(Object o, Long defaultValue) {
+        if (null == o) {
+            return defaultValue;
+        }
+
+        if ( o instanceof Long ||
+                o instanceof Integer ||
+                o instanceof Short ||
+                o instanceof Byte) {
+            return ((Number) o).longValue();
+        } else if (o instanceof Double) {
+            final long l = (Long) o;
```

Comment:
This is not going to work here as a Long can never be larger then MAX_VALUE for a long or smaller then MIN_VALUE for a Long. Technically I think there may be a bug in getInt if the number is larger then a Long can hold.

Reply:
yes my bad.  I think its ok to throw exception for numbers larger than Long. 
I dont see the bug in getInt